cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -67,8 +79,8 @@
   private final long logicalStartTime;
   private final ProgramWorkflowRunnerFactory workflowProgramRunnerFactory;
   private NettyHttpService httpService;
-  private volatile boolean running;
-  private volatile WorkflowStatus workflowStatus;
+  private ExecutorService executor;
```

Comment:
What is this for? This service is already an `AbstractExecutionThreadService`, which mean the `run()` method is already executing in a separate thread.